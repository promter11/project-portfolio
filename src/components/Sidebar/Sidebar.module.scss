@import "../../styles/_variables";
@import "../../styles/_mixins";

.sidebar {
  position: fixed;
  top: 0;
  left: 0;
  z-index: 5;
  @include flex(nowrap, column, space-between, inherit);
  @include width-height(32rem, 100vh);
  padding: 4rem;
  background-color: $sidebar-background-color;

  &:before {
    content: "";
    position: fixed;
    top: 0;
    left: 0;
    z-index: -5;
    @include width-height(100vw, 100vh);
    visibility: hidden;
    opacity: 0;
    background-color: $black-color;
  }

  @media (max-width: 576px) {
    width: 28rem;
    left: -28rem;
  }

  @media (min-width: 577px) and (max-width: 992px) {
    left: -32rem;
  }

  &.sidebarEnter,
  &.sidebarEnterActive,
  &.sidebarEnterDone,
  &.sidebarExit,
  &.sidebarExitActive,
  &.sidebarExitDone {
    transition: left 300ms;

    &:before {
      transition: opacity 300ms;
    }
  }

  &.sidebarEnter,
  &.sidebarExitActive,
  &.sidebarExitDone {
    &:before {
      visibility: hidden;
      opacity: 0;
    }

    @media (max-width: 576px) {
      left: -28rem;
    }

    @media (min-width: 577px) and (max-width: 992px) {
      left: -32rem;
    }
  }

  &.sidebarEnterActive,
  &.sidebarEnterDone,
  &.sidebarExit {
    &:before {
      visibility: visible;
      opacity: 0.5;

      @media (min-width: 993px) {
        visibility: hidden;
        opacity: 0;
      }
    }

    @media (max-width: 992px) {
      left: 0;
    }
  }

  .wrapper {
    @include flex(nowrap, row, space-between, center);
  }
}
