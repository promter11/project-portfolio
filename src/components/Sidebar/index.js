import React from "react";

import User from "../User";
import Menu from "../Menu";
import Flex from "../Flex";
import Social from "../Social";
import Language from "../Language";

import Avatar from "../../assets/images/Avatar.jpg";
import Home from "../../assets/icons/aside/home.svg";
import About from "../../assets/icons/aside/about.svg";
import Skills from "../../assets/icons/aside/skills.svg";
import Portfolio from "../../assets/icons/aside/portfolio.svg";
import Contact from "../../assets/icons/aside/contact.svg";
import { ReactComponent as Telegram } from "../../assets/icons/aside/telegram.svg";
import { ReactComponent as Github } from "../../assets/icons/aside/github.svg";
import { ReactComponent as Email } from "../../assets/icons/aside/email.svg";
import Russia from "../../assets/icons/aside/russia.svg";
import UK from "../../assets/icons/aside/uk.svg";
import styles from "./Sidebar.module.scss";

const initialUserState = {
  name: "Alexander Samylin",
  position: "Front-End Developer",
  avatar: Avatar,
};
const initialMenuState = [
  { id: 0, title: "Home", path: "/", icon: Home },
  { id: 1, title: "About", path: "/about", icon: About },
  { id: 2, title: "Skills", path: "/skills", icon: Skills },
  { id: 3, title: "Portfolio", path: "/portfolio", icon: Portfolio },
  { id: 4, title: "Contact", path: "/contact", icon: Contact },
];
const initialSocialState = [
  {
    id: 0,
    path: "https://t.me/darklurkerrr",
    component: Telegram,
  },
  {
    id: 1,
    path: "https://github.com/promter11",
    component: Github,
  },
  {
    id: 2,
    path: "mailto:promter11@gmail.com",
    component: Email,
  },
];
const initialLanguageState = [
  { id: 0, icon: Russia, active: false },
  { id: 1, icon: UK, active: true },
];

const Sidebar = () => {
  return (
    <aside className={styles.sidebar}>
      <User {...initialUserState} />
      <Menu items={initialMenuState} />
      <Flex container justifyContent="space-between" alignItems="center">
        <Language languages={initialLanguageState} />
        <Social socials={initialSocialState} />
      </Flex>
    </aside>
  );
};

export default Sidebar;
