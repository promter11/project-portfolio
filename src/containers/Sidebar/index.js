import { bindActionCreators } from "redux";
import { connect } from "react-redux";

import { getSidebarStatus } from "../../store/selectors/SidebarSelectors";
import { getLanguage } from "../../store/selectors/LanguageSelectors";
import * as SidebarActions from "../../store/actions/SidebarActions";

import Sidebar from "../../components/Sidebar";

const mapStateToProps = (state) => {
  return {
    state: {
      sidebarStatus: getSidebarStatus(state),
      language: getLanguage(state),
    },
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    actions: bindActionCreators(SidebarActions, dispatch),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Sidebar);
