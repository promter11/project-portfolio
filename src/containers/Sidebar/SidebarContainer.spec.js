import configureStore from "redux-mock-store";
import { shallow } from "enzyme";

import Sidebar from "./index";

const configureMockStore = configureStore();

const initialState = {
  LanguageReducer: {
    defaultLanguage: {
      language: "RU",
      icon: "russia.svg",
    },
  },
  SidebarReducer: {
    sidebarStatus: false,
  },
};

const store = configureMockStore(initialState);

describe("Sidebar container", () => {
  it("should renders correctly with props", () => {
    const wrapper = shallow(<Sidebar store={store} />).dive();

    expect(wrapper).toMatchSnapshot();
  });
});
