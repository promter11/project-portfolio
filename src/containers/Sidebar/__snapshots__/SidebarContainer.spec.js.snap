// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar container should renders correctly with props 1`] = `
<Sidebar
  actions={
    Object {
      "toggleStatus": [Function],
    }
  }
  state={
    Object {
      "language": "RU",
      "sidebarStatus": false,
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
